pandas/tests/io/formats/test_info.py

```diff
@@ -51,447 +51,590 @@ def datetime_frame():
-    df.info(buf=buf, memory_usage=False)
-    res = buf.getvalue().splitlines()
-    assert "memory usage: " not in res[-1]
-
-    df.info(buf=buf, memory_usage=True)
-    res = buf.getvalue().splitlines()
-
-    # memory usage is a lower bound, so print it as XYZ+ MB
-    assert re.match(r"memory usage: [^+]+\+", res[-1])
-
-    df.iloc[:, :5].info(buf=buf, memory_usage=True)
-    res = buf.getvalue().splitlines()
-
-    # excluded column with object dtype, so estimate is accurate
-    assert not re.match(r"memory usage: [^+]+\+", res[-1])
-
-    # Test a DataFrame with duplicate columns
-    dtypes = ["int64", "int64", "int64", "float64"]
-    data = {}
-    n = 100
-    for i, dtype in enumerate(dtypes):
-        data[i] = np.random.randint(2, size=n).astype(dtype)
-    df = DataFrame(data)
-    df.columns = dtypes
-
-    df_with_object_index = DataFrame({"a": [1]}, index=["foo"])
-    df_with_object_index.info(buf=buf, memory_usage=True)
-    res = buf.getvalue().splitlines()
-    assert re.match(r"memory usage: [^+]+\+", res[-1])
-
-    df_with_object_index.info(buf=buf, memory_usage="deep")
-    res = buf.getvalue().splitlines()
-    assert re.match(r"memory usage: [^+]+$", res[-1])
-
-    # Ensure df size is as expected
-    # (cols * rows * bytes) + index size
-    df_size = df.memory_usage().sum()
-    exp_size = len(dtypes) * n * 8 + df.index.nbytes
-    assert df_size == exp_size
-
-    # Ensure number of cols in memory_usage is the same as df
-    size_df = np.size(df.columns.values) + 1  # index=True; default
-    assert size_df == np.size(df.memory_usage())
-
-    # assert deep works only on object
-    assert df.memory_usage().sum() == df.memory_usage(deep=True).sum()
-
-    # test for validity
-    DataFrame(1, index=["a"], columns=["A"]).memory_usage(index=True)
-    DataFrame(1, index=["a"], columns=["A"]).index.nbytes
-    df = DataFrame(
-        data=1, index=MultiIndex.from_product([["a"], range(1000)]), columns=["A"]
-    )
-    df.index.nbytes
-    df.memory_usage(index=True)
-    df.index.values.nbytes
-
-    mem = df.memory_usage(deep=True).sum()
-    assert mem > 0
-
-
-@pytest.mark.skipif(PYPY, reason="on PyPy deep=True doesn't change result")
-def test_info_memory_usage_deep_not_pypy():
-    df_with_object_index = DataFrame({"a": [1]}, index=["foo"])
-    assert (
-        df_with_object_index.memory_usage(index=True, deep=True).sum()
-        > df_with_object_index.memory_usage(index=True).sum()
-    )
-
-    df_object = DataFrame({"a": ["a"]})
-    assert df_object.memory_usage(deep=True).sum() > df_object.memory_usage().sum()
-
-
-@pytest.mark.skipif(not PYPY, reason="on PyPy deep=True does not change result")
-def test_info_memory_usage_deep_pypy():
-    df_with_object_index = DataFrame({"a": [1]}, index=["foo"])
-    assert (
-        df_with_object_index.memory_usage(index=True, deep=True).sum()
-        == df_with_object_index.memory_usage(index=True).sum()
+class TestDataFrameInfo:
```

Comment:
so this diff is super confusing. I would rather simply make 2 test files then jamming them in one. (you can also make a sub-module if that works better).

Reply:
I created a separate module ``tests/io/formats/tests_series_info.py``.
After this PR, we can probably move both info modules to ``tests/*/methods/``.